### ISSUE TITLE
Minor suggestion about using `compat`

### ISSUE THREAD

**Issue by alphapapa** [jimporter/urgrep#4]
Hi Jim,

Well, how about that, another Emacser named Porter.  :)  I saw your post about `urgrep` on emacs-devel.  Looks very interesting.  Quick suggestion, rather than conditionally defining some functions from later Emacs versions, I think they should generally be available in the `compat` library on ELPA; so if you depend on that package and `require` it, you should be able to just call the expected functions directly.

**Comment by jimporter** [comment]
> Well, how about that, another Emacser named Porter. :)

Ah ha, I just now connected your name on the emacs-devel list to your Github username! (And to add extra confusion with me on the mailing list, there's also J.P., the ERC maintainer...)

> Quick suggestion, rather than conditionally defining some functions from later Emacs versions, I think they should generally be available in the `compat` library on ELPA; so if you depend on that package and `require` it, you should be able to just call the expected functions directly.

Thanks for the suggestion. I'd actually considered doing this a while back, but then couldn't come up with a good solution for installing the package dependencies in CI and promptly forgot about the whole thing. After some digging, I found [makel](https://github.com/DamienCassou/makel/), which provided the necessary inspiration for a stupid-simple way of doing that.
